pkgs/development/compilers/rust/rustc.nix

```
@@ -133,6 +136,11 @@ in stdenv.mkDerivation rec {
 
     # Useful debugging parameter
     # export VERBOSE=1
+  '' + lib.optionalString stdenv.isDarwin ''
+    # jemalloc's configure script misdetects the virtual address size on Darwin under Rosetta 2.
+    # See https://github.com/jemalloc/jemalloc/issues/1997
+    # Using a value of 48 should work on both aarch64-darwin and x86_64-darwin.
```

We only use jemalloc on x86_64-darwin as of now (we really should do it on all platforms like upstream does, but that's a discussion for another day), maybe we should do this for consistency/clarity?

```suggestion
  '' + lib.optionalString stdenv.isDarwin && stdenv.isx86_64 ''
    # jemalloc's configure script misdetects the virtual address size on Darwin under Rosetta 2.
    # See https://github.com/jemalloc/jemalloc/issues/1997
    # Using a value of 48 should work on both emulated and native x86_64-darwin.
```

---

done